shared/src/dto/vehicle/vehicle.dto.ts

```diff
@@ -1,65 +1,77 @@
-import { ApiProperty, PartialType } from '@nestjs/swagger';
-import { IsNotEmpty, IsOptional, ValidateNested } from 'class-validator';
-import { Type } from 'class-transformer';
+import { ApiProperty, ApiPropertyOptional, PartialType } from '@nestjs/swagger';
+import {
+  IsDate,
+  IsNotEmpty,
+  IsOptional,
+  IsString,
+  ValidateNested,
+} from 'class-validator';
+import { Transform, Type } from 'class-transformer';
 
 export class DtoVehicle {
   @ApiProperty()
-  @IsNotEmpty()
   id: number;
 
   @ApiProperty()
-  @IsNotEmpty()
   name: string;
 
   @ApiProperty()
-  @IsNotEmpty()
   licensePlate: string;
 
   @ApiProperty()
-  @IsNotEmpty()
   vin: string;
 
   @ApiProperty()
-  @IsNotEmpty()
   dateOfNextCheck: ApiDate;
 
   @ApiProperty({ nullable: true })
-  @IsOptional()
   photo: string | null;
 
   @ApiProperty({ nullable: true })
-  @IsOptional()
   additionalDetails: string | null;
 
   @ApiProperty({ nullable: true })
-  @IsOptional()
   notes: string | null;
 }
 
 export class DtoCreateVehicle {
   @ApiProperty()
   @IsNotEmpty()
+  @IsString()
   name: string;
 
   @ApiProperty()
   @IsNotEmpty()
+  @IsString()
   licensePlate: string;
 
   @ApiProperty()
   @IsNotEmpty()
+  @IsString()
   vin: string;
 
-  @ApiProperty()
+  @ApiProperty({ type: 'string', format: 'YYYY-mm-DDTHH:mm:ss.SZ' })
   @IsNotEmpty()
+  @IsDate()
+  @Type(() => Date)
```

Comment:
Shouldn't it be ApiDate as below?

Reply:
So... `ApiDate` is used as a type only and cannot be instantiated. Here I need an object that can be instantiated, so I have to use the actual `Date` object.
Also this will be used only on the server side as on the FE we only use the type information. As so we don't need to differentiate what ApiDate is on FE (string) or on the BE (date)